Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/service/impl/ConceptMergeService.java

```diff
@@ -0,0 +1,254 @@
+    }
+
+    public void getConceptEntryFromConceptMergeBean(ConceptEntry entry, ConceptsMergeBean conceptMergeBean) {
+        entry.setPos(conceptMergeBean.getSelectedPosValue());
+        entry.setConceptList(conceptMergeBean.getSelectedListName());
+        entry.setTypeId(conceptMergeBean.getSelectedTypeId());
+
+        String prefix = ",";
+        String words = conceptMergeBean.getWords().stream().map(i -> i.toString()).collect(Collectors.joining(prefix));
+        entry.setWord(words);
+
+        String description = conceptMergeBean.getDescriptions().stream().map(i -> i.toString())
+                .collect(Collectors.joining(prefix));
+        entry.setDescription(description);
+
+        String synonymIds = conceptMergeBean.getSynonymsids().stream().map(i -> i.toString())
+                .collect(Collectors.joining(prefix));
+        entry.setSynonymIds(synonymIds);
+
+        String equals = conceptMergeBean.getEqualsValues().stream().map(i -> i.toString())
+                .collect(Collectors.joining(prefix));
+        entry.setEqualTo(equals);
+
+        String similar = conceptMergeBean.getSimilarValues().stream().map(i -> i.toString())
+                .collect(Collectors.joining(prefix));
+        entry.setSimilarTo(similar);
+
+        String wordnetIds = conceptMergeBean.getWordnetIds().stream().map(i -> i.toString())
+                .collect(Collectors.joining(prefix));
+        entry.setWordnetId(wordnetIds);
+
+        addAlternativeIdsToEntry(entry, conceptMergeBean);
+        addMergedIdsToEntry(entry, conceptMergeBean);
+    }
+
+    private void addWordNetIdsToEntry(ConceptEntry entry, ConceptsMergeBean conceptMergeBean) {
+        StringBuilder wordnetBuilder = new StringBuilder();
+        String prefix = "";
+        for (String conceptId : conceptMergeBean.getConceptIds()) {
+            wordnetBuilder.append(prefix);
+            wordnetBuilder.append(conceptId);
+            prefix = ",";
+        }
+        entry.setWordnetId(wordnetBuilder.toString());
+    }
+
+    private void addAlternativeIdsToEntry(ConceptEntry entry, ConceptsMergeBean conceptMergeBean) {
+        List<String> conceptIds = conceptMergeBean.getConceptIds();
+        String selectedId = conceptMergeBean.getSelectedConceptId();
+        Set<String> alternativeIds = new HashSet<>();
+
+        for (String conceptId : conceptIds) {
+            if (!conceptId.equalsIgnoreCase(selectedId) || ConceptTypes.SPECIFIC_WORDNET_CONCEPT == conceptTypesService
+                    .getConceptTypeByConceptId(selectedId)) {
+                alternativeIds.add(conceptId);
+                ConceptEntry conceptEntry = conceptManager.getConceptEntry(conceptId);
+                alternativeIds.addAll(conceptEntry.getAlternativeIds());
+            }
+        }
+        entry.setAlternativeIds(alternativeIds);
+    }
+
+    private void addMergedIdsToEntry(ConceptEntry entry, ConceptsMergeBean conceptMergeBean) {
```

Comment:
you can replace this method probably with a java 8 stream -> filter -> map -> collect as well (just add a filter).